teerex/src/tests/tests.rs

```diff
@@ -689,17 +696,253 @@ fn confirm_proposed_sidechain_block_from_shard_neq_mrenclave_errs() {
 		));
 		assert_eq!(Teerex::enclave_count(), 1);
 
+		let header = SidechainHeader {
+			parent_hash: block_hash,
+			block_number: 1,
+			shard_id: shard4,
+			block_data_hash: block_hash,
+		};
+
 		assert_err!(
 			Teerex::confirm_proposed_sidechain_block(
 				Origin::signed(signer7.clone()),
 				shard4.clone(),
-				block_hash.clone(),
+				header,
 			),
 			Error::<Test>::WrongMrenclaveForShard
 		);
 	})
 }
 
+#[test]
+fn confirm_proposed_sidechain_block_correct_order() {
+	new_test_ext().execute_with(|| {
+		Timestamp::set_timestamp(TEST7_TIMESTAMP);
+		let signer7 = get_signer(TEST7_SIGNER_PUB);
+		let shard7 = H256::from_slice(&TEST7_MRENCLAVE);
+
+		//Ensure that enclave is registered
+		assert_ok!(Teerex::register_enclave(
+			Origin::signed(signer7.clone()),
+			TEST7_CERT.to_vec(),
+			URL.to_vec(),
+		));
+		assert_eq!(Teerex::enclave_count(), 1);
```

Comment:
This seems to be a pattern that is repeated for each test. Can we extract this to a common function?